upup/pkg/fi/cloudup/new_cluster.go

```diff
@@ -934,6 +936,17 @@ func setupTopology(opt *NewClusterOptions, cluster *api.Cluster, allZones sets.S
 			cluster.Spec.Subnets[i].Type = api.SubnetTypePublic
 		}
 
+		if opt.IPv6 {
+			if api.CloudProviderID(cluster.Spec.CloudProvider) == api.CloudProviderAWS {
+				klog.Warningf("IPv6 support is EXPERIMENTAL and can be changed or removed at any time in the future!!!")
+				for i := range cluster.Spec.Subnets {
+					cluster.Spec.Subnets[i].IPv6CIDR = fmt.Sprintf("/64#%x", i)
```

Comment:
We should reserve `/64#0` for the ClusterCIDR.

Reply:
Sounds good to me, will change.

Reply:
For future reference, read `NonMasqueradeCIDR` instead of `ClusterCIDR`. The latter is carved out of the former.

Reply:
Yes, got it, though, we should discuss (again) about it. `NonMasqueradeCIDR` may go away soon as it's part of the dockershim implementation IIRC. Seems used just by the kubenet family.

Reply:
The main use that is relevant here is that it's what `populateClusterSpec.assignSubnets()` carves the default `ClusterCIDR` and `ServiceClusterIPRange` out from.